### PR TITLE
Add tag config panel

### PR DESCRIPTION
**Pull request by balloob**
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/data/automation.ts [hunk 1]
```diff
@@ -90,6 +90,11 @@ export interface ZoneTrigger {
   event: "enter" | "leave";
 }
 
+export interface TagTrigger {
+  platform: "tag";
+  tag_id: string;
```
Comment (balloob) [inline src/data/automation.ts:95]:
```suggestion
  tag_id: string;
  device_id?: string;
```

File: src/panels/config/automation/trigger/types/ha-automation-trigger-tag.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  @property() public trigger!: TagTrigger;
+
+  public static get defaultConfig() {
+    return { tag_id: "" };
+  }
+
+  protected render() {
+    const { tag_id } = this.trigger;
+    return html`
+      <paper-input
```
Comment (balloob) [inline src/panels/config/automation/trigger/types/ha-automation-trigger-tag.ts:23]: We should make this a dropdown with all possible tags.

Comment (bramkragten) [reply src/panels/config/automation/trigger/types/ha-automation-trigger-tag.ts:23]: Yes, wanted to keep it simple for now

File: src/panels/config/tags/dialog-tag-detail.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+            <paper-input
+              dialogInitialFocus
+              .value=${this._name}
+              .configValue=${"name"}
+              @value-changed=${this._valueChanged}
+              .label="${this.hass!.localize(
+                "ui.panel.config.tags.detail.name"
+              )}"
+              .errorMessage="${this.hass!.localize(
+                "ui.panel.config.tags.detail.required_error_msg"
```
Comment (balloob) [inline src/panels/config/tags/dialog-tag-detail.ts:90]: Is name required ?

Comment (bramkragten) [reply src/panels/config/tags/dialog-tag-detail.ts:90]: Yes... with min length 1

File: src/panels/config/tags/ha-config-tags.ts [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+  protected hassSubscribe() {
+    return [
+      this.hass.connection.subscribeEvents<TagScannedEvent>((ev) => {
+        const foundTag = this._tags.find((tag) => tag.id === ev.data.tag_id);
+        if (!foundTag) {
+          this._fetchTags();
+          return;
+        }
+        foundTag.last_scanned = ev.time_fired;
+        this._tags = [...this._tags];
```
Comment (balloob) [inline src/panels/config/tags/ha-config-tags.ts:174]: Why are we copying it and not just requestUpdate ?

Comment (bramkragten) [reply src/panels/config/tags/ha-config-tags.ts:174]: Because of memoize

**Pull request merged**